src/query/storage/m3/storage.go

```diff
@@ -123,6 +121,15 @@ func (s *m3storage) FetchProm(
 		enforcer = cost.NoopChainedEnforcer()
 	}
 
+	if options.IncludeResolution {
+		resolutions := make([]int64, 0, len(attrs))
+		for _, attr := range attrs {
+			resolutions = append(resolutions, int64(attr.Resolution))
+		}
+
+		result.Metadata.Resolutions = resolutions
```

Comment:
Caught up briefly, was wondering if we take `.Metadata` out of `PromResult` completely if the remote read path does not take resolutions.

Reply:
Summarizing talk here in case we need future reference:

Metadata is primarily used to track warnings through the series lifetime outside of functions, so for stuff like hitting the m3 limit or fetch warnings, so even in the prom read path it's necessary.